cpp/src/strings/copying/concatenate.cu

```diff
@@ -287,12 +285,15 @@ std::unique_ptr<column> concatenate(host_span<column_view const> columns,
         column_view offsets_child = column->child(strings_column_view::offsets_column_index);
         column_view chars_child   = column->child(strings_column_view::chars_column_index);
 
-        auto d_offsets       = offsets_child.data<int32_t>() + column_offset;
-        int32_t bytes_offset = thrust::device_pointer_cast(d_offsets)[0];
+        auto bytes_offset =
+          cudf::detail::get_value<offset_type>(offsets_child, column_offset, stream);
 
         // copy the chars column data
-        auto d_chars    = chars_child.data<char>() + bytes_offset;
-        size_type bytes = thrust::device_pointer_cast(d_offsets)[column_size] - bytes_offset;
+        auto d_chars = chars_child.data<char>() + bytes_offset;
+        auto const bytes =
+          cudf::detail::get_value<offset_type>(offsets_child, column_size + column_offset, stream) -
+          bytes_offset;
```

Comment:
This part of the code, overlooks the size of `chars_child` column. while debugging a memory error, I found that `chars_column.size()` might be less than `bytes` and it caused `cudaMemcpyAsync` invalid argument failure.

Probably a `CUDA_EXPECTS` might satisfy size checking.

Reply:
It is easy enough to add the `CUDF_EXPECTS` but the `cudaMemcpyAsync` is wrapped in a `CUDA_TRY` so an exception is thrown anyway.